astropy/units/format/latex.py

```diff
@@ -103,7 +103,8 @@ def format_exponential_notation(cls, val):
             The value in exponential notation in a format suitable for LaTeX.
         """
         if np.isfinite(val):
-            m, ex = utils.split_mantissa_exponent(val)
+            m, ex = utils.split_mantissa_exponent(val,
+                np.get_printoptions()['precision'])
```

Comment:
With the change below, this would be
```
format_spec='.{0}g'.format(np.get_printoptions()['precision'])
```

Though note that this change would also imply the default precision for printing a scale of a unit would start to depend on `np.printoptions` - which is a bit odd, since the scale is not a numpy array. If we want to keep the change limited to quantities, one should add the `format_spec` argument here too, and only set it in `_repr_latex_` - I think I'm in favour of that.